test/units/modules/test_pip.py

```diff
@@ -1,40 +1,171 @@
+        ['django>1.11.1,<1.11.2,ipaddress', 'simpleproject<2.0.0,>1.1.0'],
+        ['django>1.11.1', '<1.11.2', 'ipaddress,simpleproject<2.0.0,>1.1.0'],
+    ),
+)
+def test_recover_package_name(test_input):
+    from ansible.modules import pip
+    expected = [
+        'django>1.11.1,<1.11.2',
+        'ipaddress',
+        'simpleproject<2.0.0,>1.1.0',
+    ]
     assert pip._recover_package_name(test_input) == expected
+
+
+@pytest.mark.parametrize(
+    'test_input',
+    (
+        ['django>1.11.1', '<1.11.2', 'ipaddress', 'simpleproject<2.0.0', '>1.1.0'],
+        ['django>1.11.1,<1.11.2,ipaddress', 'simpleproject<2.0.0,>1.1.0'],
+        ['django>1.11.1', '<1.11.2', 'ipaddress,simpleproject<2.0.0,>1.1.0'],
+    ),
+)
+def test_recover_package_name_v2(test_input, run_ansible_module):
+    mod_res = run_ansible_module(name=test_input)
+    expected = [
+        'django<1.11.2,>1.11.1',
+        'ipaddress',
+        'simpleproject<2.0.0,>1.1.0',
+    ]
+    assert mod_res['cmd'][-len(expected):] == expected
+
+
+def test_new_mod_make(make_ansible_module_caller):
+    # run_module = make_ansible_module_caller('pip')
+    # run_module = make_ansible_module_caller('ansible_collections.ansible.builtin.pip')
+    # run_module = make_ansible_module_caller(
+    #     'ansible_collections.community.general.modules.pip_package_info',
+    # )
+    # run_module = make_ansible_module_caller(
+    #     'ansible_collections.community.general.pip_package_info',
+    # )
+    run_module = make_ansible_module_caller('ansible.modules.pip')
+    assert run_module(name='bottle')['cmd'][1:] == [
+        '-m', 'pip.__main__', 'install', 'bottle',
+    ]
+    assert run_module(name='bottle')  # the second invocation doesn't 💣
+
+
+@pytest.fixture
+def patched_ansible_module_class(patched_ansible_module_class):
+    class AnsibleModule(patched_ansible_module_class):
+        def run_command(self, *args, **kwargs):
+            if args[0][-2:] == ['install', 'bottle']:
+                return 666, '<out>', '<err>'
+            return super(AnsibleModule, self).run_command(*args, **kwargs)
+
+    return AnsibleModule
+
+
+def test_new_mod_global_fail(run_ansible_module):
+    failed_response = run_ansible_module(name='bottle')
+    print(failed_response)
+    assert failed_response['failed']
+
+
+@pytest.mark.parametrize(
+    'ansible_module_name',
+    ('ansible.modules.pip', ),
+    indirect=('ansible_module_name', ),
+)
+def test_new_mod_param(run_ansible_module):
+    assert run_ansible_module(name='bottle', executable='pip3.9')['cmd'][1:] == ['install', 'bottle']
+
+
+@pytest.mark.parametrize(
+    'yaml_task',
+    (
+        # NOTE: FQCN cannot be used here because ansible-test only puts
+        # NOTE: `ansible.modules` and `ansible.module_utils` in the venv.
+        # These don't work:
+        # * ansible_collections.ansible.legacy.pip
+        # * ansible_collections.ansible.builtin.pip
```

Comment:
@mattclay I wonder if it's possible to do something about this

Reply:
Is there a reason you'd like to involve the collection loader here? That doesn't seem like something module unit tests should be concerned with.

Reply:
@mattclay well, this whole effort is me experimenting with improving the UX/DX. And I feel like allowing the collection authors use a similar format to what would be used in playbooks, contributes to a better experience.

Forcing folks to use a different importable path (since it's `runpy` under the hood) seems like a way to make this testing interface foreign, which is the opposite of what I intend it to be.